packages/headless/src/features/commerce/common/base-commerce-api-request-builder.ts

```diff
@@ -30,7 +30,7 @@ export const buildBaseCommerceAPIRequest = (
         state.configuration.environment
       ),
     organizationId: state.configuration.organizationId,
-    trackingId: state.configuration.analytics.trackingId,
+    trackingId: state.configuration.analytics.trackingId!,
```

Comment:
@samisayegh 
Only change affecting commerce directly here.

After your suggestion, the state `state.configuration.analytics.trackingId` is initialized as an empty string.

And `tackingId` is still required in the configuration state under commerce/

So we should be good with the `!` only.

Reply:
Just to make sure, is the `!` needed because the type of `trackingId` is inherited from [this interface](https://github.com/coveo/ui-kit/pull/5316/files#diff-a4f8bb3532fc4170d823e08a8962590441a17be552d7c73f897e3e3c19ad9b5eR178) which sets it as optional?

Reply:
Correct, since the type for the `BaseCommerceAPIRequest` marks the `trackingId` as a non-optional string, and the `state.configuration.analytics.trackingId` is essentially `string | undefined`, this tells TypeScript that we're sure at this moment that we'll have validated that `trackingId` is in fact not undefined.